modules/agent/main.tf

```diff
@@ -18,14 +18,16 @@ module "agent" {
   config_drive       = var.node_config.config_drive
   floating_ip_pool   = var.node_config.floating_ip_pool
   user_data          = var.node_config.user_data
-  boot_from_volume   = var.node_config.boot_from_volume
-  boot_volume_size   = var.node_config.boot_volume_size
+  boot_from_volume   = var.boot_from_volume
+  boot_volume_size   = var.boot_volume_size
+  boot_volume_type   = var.boot_volume_type
```

Comment:
Do we really want to have controplane not booting from volume but agents booting from ?

Reply:
If I remove these my Nvidia example gives this error
```
terraform plan
╷
│ Error: Missing required argument
│
│   on ../terraform-openstack-rke2-test/modules/agent/main.tf line 1, in module "agent":
│    1: module "agent" {
│
│ The argument "boot_from_volume" is required, but no definition was found.
╵
╷
│ Error: Missing required argument
│
│   on ../terraform-openstack-rke2-test/modules/agent/main.tf line 1, in module "agent":
│    1: module "agent" {
│
│ The argument "boot_volume_size" is required, but no definition was found.
╵
╷
│ Error: Missing required argument
│
│   on ../terraform-openstack-rke2-test/modules/agent/main.tf line 1, in module "agent":
│    1: module "agent" {
│
│ The argument "boot_volume_type" is required, but no definition was found.

```

Reply:
I see.
Please just set default values in `modules/agent/variables.tf` to makes this optional :

```tf
variable "boot_from_volume" {
  type = bool
  default     = false
}

variable "boot_volume_size" {
  type = number
  default     = 20
}

variable "boot_volume_type" {
  type = string
  default     = ""
}
```

Reply:
thanks for the clarification , all done